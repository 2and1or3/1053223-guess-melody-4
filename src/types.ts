enum GameType {
  ARTIST = `artist`,
  GENRE = `genre`,
}

interface GenreAnswer {
  genre: string;
  src: string;
}

interface ArtistAnswer {
  artist: string;
  pictureSrc: string;
}

interface GenreQuestion {
  type: GameType.GENRE;
  genre: string;
  answers: Array<GenreAnswer>;
}

interface ArtistQuestion {
  type: GameType.ARTIST;
  artist: string;
  trackSrc: string;
  answers: Array<ArtistAnswer>;
}

export { GenreAnswer, ArtistAnswer, GenreQuestion, ArtistQuestion, GameType };
