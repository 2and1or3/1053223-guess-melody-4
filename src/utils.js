import {GameType} from './types.ts';

const mockAudioNode = (element) => {
  if (element.type === `audio`) {
    return element;
  }

  return null;
};

const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const adapterForGenreType = (question) => question;

const adapterForArtistType = (question) => ({
  type: question.type,
  artist: question.song.artist,
  trackSrc: question.song.src,
  answers: question.answers.map((answer) => ({
    artist: answer.artist,
    pictureSrc: answer.picture,
  }))
});

const adapterToLocalQuestions = (questions) => {

  const localQuestions = questions.map((question) => {
    switch (question.type) {
      case (GameType.ARTIST):
        return adapterForArtistType(question);
      case (GameType.GENRE):
        return adapterForGenreType(question);

      default:
        return question;
    }
  });

  return localQuestions;
};

export {mockAudioNode, adapterToLocalQuestions, extend};
