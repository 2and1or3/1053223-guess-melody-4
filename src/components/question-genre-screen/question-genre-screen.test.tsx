import * as React from "react";
import * as renderer from "react-test-renderer";

import QuestionGenreScreen from './question-genre-screen';
import {GenreQuestion, GameType} from '../../types';
import {noop} from '../../utils';

const question: GenreQuestion = {
  type: GameType.GENRE,
  genre: `rock`,
  answers: [
    {
      genre: `rock`,
      src: `src-for-track-1`,
    },
    {
      genre: `jazz`,
      src: `src-for-track-2`,
    },
    {
      genre: `pop`,
      src: `src-for-track-3`,
    },
    {
      genre: `alternative`,
      src: `src-for-track-4`,
    }
  ]
};

it(`Render QuestionGenreScreen component`, () => {
  const tree = renderer
    .create(<QuestionGenreScreen
      question = {question}
      renderPlayer = {noop}
      userAnswers = {new Array(question.answers.length).fill(false)}
      onSubmit = {noop}
      onChange = {noop}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
