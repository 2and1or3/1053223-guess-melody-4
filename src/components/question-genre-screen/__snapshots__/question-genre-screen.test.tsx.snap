// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render QuestionGenreScreen component 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите 
    rock
     треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="rock0"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="rock"
        />
        <label
          className="game__check"
          htmlFor="rock0"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="jazz1"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="jazz"
        />
        <label
          className="game__check"
          htmlFor="jazz1"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="pop2"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="pop"
        />
        <label
          className="game__check"
          htmlFor="pop2"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          checked={false}
          className="game__input visually-hidden"
          id="alternative3"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="alternative"
        />
        <label
          className="game__check"
          htmlFor="alternative3"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
